docs/decisions/0001-modular-frontend.rst

```diff
@@ -0,0 +1,68 @@
+# Modular frontends
+
+## Background
+
+A couple of issues have been identified with the current MFE architecure that
+we hope to resolve with this proposal. Those issues are:
+
+- Duplicated javascript and styling code across multiple MFEs that needs to be
+  loaded multiple times.
+- Large bundle size for each MFE that includes the entire codebase instead of
+  the smaller subset of features needed.
+- Use of different libraries, versions of libraries etc across MFEs further
+  increasing overall bundle size.
+- An increasing number of different MFEs to keep track of, build and update.
+- Poor customisability due to the lack of plugins, or other hooks into MFEs
+- No seamless routing between MFEs.
+
+## Proposal
+
+Our proposal boils down to the following:
+
+- Creating of an LMS mono-repo that pulls in MFEs related to the LMS into a
+  single repo, potentially starting as submodules and eventully being hosted in
+  the main repo.
+- Use of webpack module federation to provide both a mechanism to support loose
+  dependencies between MFEs, and for plugins
```

Comment:
I see the rationale below but can you please explain this in more detail and include example code where helpful? Just from reading this it's unclear to me what this does or what "loose dependencies between MFEs" even means. What kind of dependencies do we have between MFEs? I naively assumed it was mostly that multiple MFEs depended on shared components/deps like frontend-platform, paragon, theme header etc. but didn't really have dependencies or coupling between the MFEs themselves, so perhaps you could unpack that?

Reply:
I definitely meant to expand this document a lot more, but was honestly out of time and wanted to have the basics ready for review before I invested more in this direction. 

By loose dependencies what I mean is that we can have MFEs check for each other's presence at runtime and include components dynamically.